analytics/scala/src/main/scala/com/koverse/example/spark/WordCountDatasetTransform.scala

```diff
@@ -16,19 +16,51 @@
 
 package com.koverse.example.spark
 
+import java.io.Serializable
+
 import com.koverse.sdk.Version
 import com.koverse.sdk.data.Parameter
 import com.koverse.sdk.transform.scala.{DatasetTransform, DatasetTransformContext}
 import org.apache.spark.sql.Dataset
 
+case class Message(var article: String, var id: Long) extends Serializable {
+
+  def this() = this("", 0)
+// article must be text and set by textFieldParam value
+  def getArticle: String = article
+
+  def setArticle(article: String): Unit = {
+    this.article = article
+  }
+
+  def getId: Long = id
+
+  def setId(id: Long): Unit = {
+    this.id = id
+  }
+}
+
+case class WordCount(var text: String, var count: Int) extends Serializable {
```

Comment:
You don't need 'var' here. Just text:String etc.

Reply:
Actually, I do need it since I added getters/setters